planner/core/task.go

```diff
@@ -951,6 +972,46 @@ func buildIndexLookUpTask(ctx sessionctx.Context, t *copTask) *rootTask {
 	return newTask
 }
 
+func extractRows(p PhysicalPlan) float64 {
+	f := float64(0)
+	for _, c := range p.Children() {
+		if len(c.Children()) != 0 {
+			f += extractRows(c)
+		} else {
+			f += c.statsInfo().RowCount
+		}
+	}
+	return f
+}
+
+func calcPagingCost(ctx sessionctx.Context, t *copTask) float64 {
+	sessVars := ctx.GetSessionVars()
+	indexRows := t.indexPlan.statsInfo().RowCount
+	expectCnt := float64(t.expectCnt)
+	sourceRows := extractRows(t.indexPlan)
+	// with paging, the scanned rows is always less than or equal to source rows.
+	expectCnt = math.Min(expectCnt, sourceRows)
+	seekCnt := float64(1)
+	if expectCnt > pagingGrowingSum {
+		// if the expectCnt is larger than pagingGrowingSum, calculate the seekCnt for the excess.
+		seekCnt += float64(int(8 + (expectCnt-pagingGrowingSum)/maxPagingSize))
```

Comment:
Do you mean that `(diff+maxPagingSize-1)/maxPagingSize` is a round-off which guarantees the result is at least 1?

Reply:
And why it's `6 + ...`, I think there will be 8 requests during growing to the `maxPagingSize`, and when it's over `pagingGrowingSum `, there should be at least 9.

Reply:
> when it's over pagingGrowingSum , there should be at least 9.
Why 9? I think it should be 8... `6 + ...` is wrong, but I think it should be `7 + ...` because the default value of `seekCnt` is 1.